nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestDocumentBlobColdStorage.java

```diff
@@ -0,0 +1,124 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Salem Aouana
+ */
+
+package org.nuxeo.ecm.core;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.nuxeo.ecm.core.FakeThumbnailFactory.FAKE_THUMBNAIL_CONTENT;
+import static org.nuxeo.ecm.core.api.impl.blob.AbstractBlob.UTF_8;
+import static org.nuxeo.ecm.core.blob.DocumentBlobColdStorageHelper.COLD_STORAGE_CONTENT_PROPERTY;
+import static org.nuxeo.ecm.core.blob.DocumentBlobColdStorageHelper.FILE_CONTENT_PROPERTY;
+import static org.nuxeo.ecm.core.blob.DocumentBlobColdStorageHelper.moveBlob;
+import static org.nuxeo.ecm.core.schema.FacetNames.COLD_STORAGE;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.core.api.Blob;
+import org.nuxeo.ecm.core.api.Blobs;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.blob.DocumentBlobColdStorageHelper;
+import org.nuxeo.ecm.core.test.CoreFeature;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.LogCaptureFeature;
+import org.nuxeo.runtime.test.runner.LogFeature;
+import org.nuxeo.runtime.test.runner.LoggerLevel;
+import org.nuxeo.runtime.test.runner.TransactionalFeature;
+
+/**
+ * @since 11.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features({ CoreFeature.class, LogFeature.class, LogCaptureFeature.class })
+@Deploy("org.nuxeo.ecm.core.test.tests:OSGI-INF/test-document-blob-coldstorage-contribs.xml")
+public class TestDocumentBlobColdStorage {
+
+    protected static final String BLOB_FILE_CONTENT = "foo";
+
+    @Inject
+    protected CoreSession session;
+
+    @Inject
+    protected TransactionalFeature transactionalFeature;
+
+    @Inject
+    protected LogCaptureFeature.Result logCaptureResult;
+
+    @Test
+    public void shouldMoveBlobDocument() throws IOException {
+        DocumentModel documentModel = session.createDocumentModel("/", "AnyFile", "File");
+        documentModel.setPropertyValue("file:content", (Serializable) Blobs.createBlob(BLOB_FILE_CONTENT));
+        documentModel = session.createDocument(documentModel);
+        assertFalse(documentModel.hasFacet(COLD_STORAGE));
+
+        // move the blob to cold storage
+        moveBlob(session, documentModel.getRef());
+        session.saveDocument(documentModel);
+        transactionalFeature.nextTransaction();
+
+        documentModel = session.getDocument(documentModel.getRef());
```

Comment:
Florent recently suggested to use `DocumentModel#refresh` instead.

Reply:
for the first test `shouldMoveBlobDocument` i will do it but the second when i replace the get document but the refresh then my test fail on: 
```
  assertEquals(String.format("The main content of document: %s is already in cold storage", documentModel),
                caughtEvents.get(0));
```
because the the expected one had in his `title` the value of `Doc Ref`

Reply:
OK so if it's not working as expected maybe not use at all here to be consistent between the 2 tests, WDYT?